src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -582,23 +584,26 @@ private static int IndexOfSessionInList(List<SessionInfo> sessions, int etwSessi
             {
 #if (!ES_BUILD_PCL && !ES_BUILD_PN && PLATFORM_WINDOWS)
                 string regKey = @"\Microsoft\Windows\CurrentVersion\Winevt\Publishers\{" + m_providerId + "}";
-                if (System.Runtime.InteropServices.Marshal.SizeOf(typeof(IntPtr)) == 8)
-                    regKey = @"HKEY_LOCAL_MACHINE\Software" + @"\Wow6432Node" + regKey;
+                if (IntPtr.Size == 8)
+                    regKey = @"Software" + @"\Wow6432Node" + regKey;
                 else
-                    regKey = @"HKEY_LOCAL_MACHINE\Software" + regKey;
+                    regKey = @"Software" + regKey;
 
                 string valueName = "ControllerData_Session_" + etwSessionId.ToString(CultureInfo.InvariantCulture);
 
                 // we need to assert this permission for partial trust scenarios
 #if !CORECLR
                 (new RegistryPermission(RegistryPermissionAccess.Read, regKey)).Assert();
 #endif
-                data = Microsoft.Win32.Registry.GetValue(regKey, valueName, null) as byte[];
-                if (data != null)
+                using (var key = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(regKey))
                 {
-                    // We only used the persisted data from the registry for updates.   
-                    command = ControllerCommand.Update;
-                    return true;
+                    data =  key?.GetValue(valueName, null) as byte[];
```

Comment:
The check for `key==null` was missing in #11009 that I believe caused the regression.